openpype/pipeline/plugin_discover.py

```diff
@@ -0,0 +1,264 @@
+import os
+import inspect
+import traceback
+
+from openpype.lib.python_module_tools import (
+    modules_from_path,
+    classes_from_module,
+)
+
+
+class DiscoverResult:
+    """Hold result of publish plugins discovery.
+
+    Stores discovered plugins duplicated plugins and file paths which
+    crashed on execution of file.
+    """
```

Comment:
We might want to be explicit about mentioning that this holds the discovered result of a single superclass type.

Also I feel like we don't need to mention "Hold result of.." but can just say: "Result of Plug-ins discovery of a single `superclass` type."

Reply:
I've copied it from new publisher. Will change it.

Reply:
By the way. Should this class maybe be implemented similar to how RenderProducts are implemented?  Would that make it "simpler" to maintain and/or more explicit that it's just a data storage?